src/deep-resource/test/Resource.js

```diff
@@ -4,180 +4,115 @@ import chai from 'chai';
-          },
-        },
-        'say-test': {
-          type: 'external',
-          methods: ['GET',],
-          region: 'us-west-2',
-          source: {
-            api: 'https://1zf47jpvxd.execute-api.us-west-2.amazonaws.com/dev/hello-world-example/sample/say-test',
-            original: 'http://petstore.swagger.io/v2/store/inventory',
-          },
-        },
-      },
-    },
-  };
   let microserviceIdentifier = 'hello.world.example';
-  let microserviceInstance = new Instance(microserviceIdentifier, testResources[microserviceIdentifier]);
+  let microserviceInstance = new Instance(microserviceIdentifier,
+    backendConfig.microservices[microserviceIdentifier].resources);
   let resource = null;
   let resourceName = 'sample';
+  let backendKernelInstance = null;
 
   test('Class Resource exists in Resource', function() {
-    resource = new Resource(testResources);
     chai.expect(typeof Resource).to.equal('function');
   });
 
-  test('Load Kernel by using Kernel.loadFromFile()', function(done) {
-    let error = null;
-    let backendKernelInstanse = null;
-
+  test('Load Kernel by using Kernel.load()', function(done) {
     let callback = (backendKernel) => {
+      chai.assert.instanceOf(
+        backendKernel, Kernel, 'backendKernel is an instance of Kernel');
+
+      backendKernelInstance = backendKernel;
       resource = backendKernel.get('resource');
-      chai.expect(backendKernel).to.be.not.eql({});
-      chai.expect(resource).to.be.not.eql({});
+
+      chai.assert.instanceOf(
+        resource, Resource, 'resource is an instance of Resource'
+      );
 
       // complete the async
       done();
     };
-    try {
-      backendKernelInstanse = new Kernel({
-        Cache: Cache,
-        Security: Security,
-        Resource: Resource,
-      }, Kernel.BACKEND_CONTEXT);
-      backendKernelInstanse.loadFromFile('./test/common/backend-cfg-json.json', callback);
-    } catch (e) {
-      error = e;
-    }
+
+    KernelFactory.create({
+      Cache: Cache,
+      Security: Security,
+      Resource: Resource,
+    }, callback);
   });
 
   test('Check constructor sets _resources', function() {
-    chai.expect(Object.keys(resource._resources)).to.be.eql(['hello.world.example', 'deep.ng.root']);
-    chai.expect(resource._resources).to.be.not.eql({});
+    chai.expect(Object.keys(resource._resources)).to.be.eql(
+      ['hello.world.example', 'deep.ng.root']
+    );
   });
 
-
   test('Check has() method returns false', function() {
-    let error = null;
-    let actualResult = null;
-    try {
-      resource.bind(microserviceInstance);
-      actualResult = resource.has('invalid_resource_identifier');
-    } catch (e) {
-      error = e;
-    }
-
-    chai.expect(error).to.be.equal(null);
-    chai.expect(actualResult).to.be.equal(false);
+    resource.bind(microserviceInstance);
+    chai.expect(resource.has('invalid_res_identifier')).to.be.equal(false);
```

Comment:
- please add more use cases for resources identifier considering this format @ms_identitifer:resource_name:action_name